src/error.jl

```diff
@@ -23,7 +24,9 @@ function gdaljl_errorhandler(class::CPLErr, errno::Cint, errmsg::Cstring)
     # function signature needs to match the one in __init__, and the signature
     # of the callback for a custom error handler in the GDAL docs
     if class in throw_class
-        throw(GDALError(class, errno, unsafe_string(errmsg)))
+        msg = unsafe_string(errmsg)
+        cplerrorreset()
```

Comment:
You call `cplerrorreset` twice now, one here, one in the `GDALError`.

Reply:
That is a different path though. That is in the 0 arg constructor, which this one doesn't use.